### PR TITLE
[KIR 327] Binding Destination

### PR DESCRIPTION
**Pull request by SelcukEdma**
Changes to be Merged:

Bind destination

NOTE:
Will need to be able to dropdown binding.

### CODE REVIEW

File: src/State/Hydration/reducers.ts [hunk 1]
```diff
@@ -4,8 +4,9 @@ export const initialState = {
   isDestinationModalOpen: false,
   isEditorOpen: false,
   selectedNode: null,
-  metaData: {
-    sourceTiles: {}
+  metadata: {
+    sourceTiles: {},
```
Comment (ed-roh) [inline src/State/Hydration/reducers.ts]: is there a reason for the naming of "sourceTiles", why not sources? just wondering

File: src/State/Hydration/api.js [hunk 2]
```diff
@@ -15,3 +15,19 @@ export const getSourceTiles = () => {
...
 };
+
+export const getDestinations = () => {
+  const request = constructRequest(
+    config.apiUrl,
+    `${config.apiPath}/users/data`,
+    'GET',
+    {
+      params: { dropdown: 'sensitivity' }
+      // params: { dropdown: 'databases', sensitivity: 'non-sensitive' }
```
Comment (ed-roh) [inline src/State/Hydration/api.js]: unneeded comment

File: src/State/Hydration/sagas.ts [hunk 3]
```diff
@@ -1,6 +1,6 @@
 import * as types from './types';
 import { takeEvery, put, call } from 'redux-saga/effects';
-import { getSourceTiles } from './api';
+import { getSourceTiles, getDestinations } from './api';
 
 export function* handleSourceTiles() {
```
Comment (ed-roh) [inline src/State/Hydration/sagas.ts]: we should probably meet to have a consistent naming structure such as prefixing this with "work" and api request naming

File: src/State/Hydration/sagas.ts [hunk 4]
```diff
@@ -14,6 +14,19 @@ export function* handleSourceTiles() {
...
+    yield put({
+      type: types.DESTINATIONS_REQUEST_SUCCESS,
+      destinations: response
+    });
+  } catch (error) {
+    yield put({ type: types.DESTINATIONS_REQUEST_FAILURE, message: error });
+  }
+}
+
 export default function* actionWatcher() {
```
Comment (ed-roh) [inline src/State/Hydration/sagas.ts]: same as above, but have "watch" as a prefix

**Review by ed-roh — COMMENTED** [review]
probably need to meet for a few discussions of consistent naming

**Comment by SelcukEdma** [comment]
@the3ddy Yeah lets sync up for naming convention

File: src/State/Hydration/api.js (hunk 2 above)
Comment (SelcukEdma) [reply src/State/Hydration/api.js]: removed

File: src/State/Hydration/sagas.ts (hunk 3 above)
Comment (SelcukEdma) [reply src/State/Hydration/sagas.ts]: added `work` as a prefix

File: src/State/Hydration/sagas.ts (hunk 4 above)
Comment (SelcukEdma) [reply src/State/Hydration/sagas.ts]: added `watch` as a prefix

File: src/State/Hydration/reducers.ts (hunk 1 above)
Comment (SelcukEdma) [reply src/State/Hydration/reducers.ts]: changed to `sources`

**Review by ed-roh — APPROVED** [review]

**Pull request merged**